src/pages/api/reroute/page.ts

```diff
@@ -0,0 +1,38 @@
+import type { APIRoute } from 'astro';
+import { datocmsRequest } from '@lib/datocms';
+import type { ReroutePageQuery, SiteLocale } from '@lib/datocms/types';
+import { getPageHref } from '@lib/routing';
+import query from './_page.query.graphql';
+
+export const prerender = false;
+
+const jsonResponse = (data: object, status: number = 200) => {
+  return new Response(JSON.stringify(data), {
+    status,
+    headers: {
+      'Content-Type': 'application/json',
+    },
+  });
+};
+
+export const GET: APIRoute = async ({ request }) => {
```

Comment:
This is a generic alternate approach that I used in the nododos-website. But I think the canonical url check in `[locale]/[...path]/index.astro` is easier and probably the way to go.

Reply:
Hmmm. I guess the canonical check works on localhost and preview as the generic page route is then always handled dynamically on the server. However the production environment has prerendered pages and will throw a 404 I suppose. So I guess we do need this reroute API handler.

In that case the config of the Preview field of the generic page will need to use that one instead.